src/Planners/src/TimeVaryingDCMPlanner.cpp

```diff
@@ -0,0 +1,749 @@
+
+        return casadi::Function("eCMP", {omega, vrp, omegaDot}, {rhs});
+    }
+
+    /**
+     * Clear the optimization problem
+     */
+    void clear()
+    {
+        opti = casadi::Opti();
+    }
+
+    /**
+     * Setup the optimization problem options
+     */
+    void setupOptiOptions()
+    {
+        casadi::Dict casadiOptions;
+        casadi::Dict ipoptOptions;
+
+        if (this->optiSettings.solverVerbosity != 0)
+        {
+            casadi_int ipoptVerbosity = static_cast<long long>(optiSettings.solverVerbosity - 1);
+            ipoptOptions["print_level"] = ipoptVerbosity;
+            casadiOptions["print_time"] = true;
+        } else
+        {
+            ipoptOptions["print_level"] = 0;
+            casadiOptions["print_time"] = false;
+        }
+        ipoptOptions["linear_solver"] = this->optiSettings.ipoptLinearSolver;
+
+        this->opti.solver("ipopt", casadiOptions, ipoptOptions);
+    }
+
+    /**
+     * Setup opti stack
+     */
+    void setupOpti(const std::size_t& horizonSampling)
+    {
+        using Sl = casadi::Slice;
+
+        auto dcmDynamicalSystem = this->dcmDynamicsIntegrator->dynamicalSystem().lock();
+
+        // define the size of the optimization variables
+        this->optiVariables.dcm = this->opti.variable(this->dcmVectorSize, horizonSampling + 1);
+        this->optiVariables.omega = this->opti.variable(1, horizonSampling + 1);
+        this->optiVariables.vrp = this->opti.variable(this->vrpVectorSize, horizonSampling);
+        this->optiVariables.omegaDot = this->opti.variable(1, horizonSampling);
+
+        // set the initial constraints
+        this->optiParameters.dcmInitialPosition = this->opti.parameter(this->dcmVectorSize);
+        this->optiParameters.dcmFinalPosition = this->opti.parameter(this->dcmVectorSize);
+        this->optiParameters.dcmInitialVelocity = this->opti.parameter(this->dcmVectorSize);
+        this->optiParameters.dcmFinalVelocity = this->opti.parameter(this->dcmVectorSize);
+        this->optiParameters.omegaInitialValue = this->opti.parameter();
+        this->optiParameters.omegaFinalValue = this->opti.parameter();
+
+        // set constraints and cost functions whose equations does not change in the optimization
+        // problem
+        auto& dcm = this->optiVariables.dcm;
+        auto& vrp = this->optiVariables.vrp;
+        auto& omega = this->optiVariables.omega;
+        auto& omegaDot = this->optiVariables.omegaDot;
+
+        // initial values
+        this->opti.subject_to(omega(Sl(), 0) == this->optiParameters.omegaInitialValue);
+        this->opti.subject_to(dcm(Sl(), 0) == this->optiParameters.dcmInitialPosition);
+
+        // initial value of the DCM dynamics
+        dcmDynamicalSystem->setState({dcm(Sl(), 0), omega(Sl(), 0)});
+        dcmDynamicalSystem->setControlInput({vrp(Sl(), 0), omegaDot(Sl(), 0)});
+
+        std::tuple<casadi::MX, casadi::MX> stateDerivative;
```

Comment:
Maybe you can use the alias here.

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/61/commits/995a9256528dab15719d481293cd049d257ac94c